lib/matplotlib/backend_bases.py

```diff
@@ -3604,21 +3610,28 @@ def _tool_triggered_cbk(self, event):
 
     def _add_tool_cbk(self, event):
         """Captures 'tool_added_event' and adds the tool to the toolbar"""
-        image = self._get_image_filename(event.tool.image)
+        tool, group, position = event.tool, event.data['group'], \
+                                                event.data['position']
+        if group is None:
+            return
```

Comment:
I agree with what you want to achieve, but I really disagree with implementing it like this.  My next PR, the one that I have been talking about, will, in my opinion give a better implementation of this.

Perhaps I should push ahead and just release my next PR as a w-i-p so that you can see for yourself.